src/main/java/org/jabref/gui/PreviewPanel.java

```diff
@@ -264,140 +221,76 @@ public BibEntry getEntry() {
-
-    private void scrollToTop() {
-        SwingUtilities.invokeLater(() -> scrollPane.getVerticalScrollBar().setValue(0));
+    private void setPreviewLabel(String text) {
+        DefaultTaskExecutor.runInJavaFXThread(() -> {
+            previewView.getEngine().loadContent(text);
+            this.setHvalue(0);
+        });
     }
 
     @Override
     public void highlightPattern(Optional<Pattern> newPattern) {
+        // TODO: Implement that search phrases are highlighted
         this.highlightPattern = newPattern;
         update();
     }
 
-    public Optional<Pattern> getHighlightPattern() {
-        return highlightPattern;
-    }
-
     /**
      * this fixes the Layout, the user cannot change it anymore. Useful for testing the styles in the settings
-     * @param parameter should be either a {@link String} (for the old PreviewStyle) or a {@link CitationStyle}.
+     *
+     * @param layout should be either a {@link String} (for the old PreviewStyle) or a {@link CitationStyle}.
      */
-    public PreviewPanel setFixedLayout(Object parameter) {
+    public void setFixedLayout(String layout) {
         this.fixedLayout = true;
-
-        if (parameter instanceof String) {
-            updatePreviewLayout((String) parameter);
-        } else if (parameter instanceof CitationStyle) {
-            layout = Optional.empty();
-            if (basePanel.isPresent()) {
-                basePanel.get().getCitationStyleCache().setCitationStyle((CitationStyle) parameter);
-            }
-        } else {
-            LOGGER.error("unknown style type");
-        }
-        update();
-        return this;
+        updatePreviewLayout(layout);
     }
 
-    class PrintAction extends AbstractAction {
-        public PrintAction() {
-            super(Localization.lang("Print entry preview"), IconTheme.JabRefIcon.PRINTED.getIcon());
-
-            putValue(Action.SHORT_DESCRIPTION, Localization.lang("Print entry preview"));
-        }
-
-        @Override
-        public void actionPerformed(ActionEvent arg0) {
-
-            // Background this, as it takes a while.
-            JabRefExecutorService.INSTANCE.execute(() -> {
-                try {
-                    PrintRequestAttributeSet pras = new HashPrintRequestAttributeSet();
-                    pras.add(new JobName(bibEntry.flatMap(BibEntry::getCiteKeyOptional).orElse("NO ENTRY"), null));
-                    previewPane.print(null, null, true, null, pras, false);
-
-                } catch (PrinterException e) {
-                    // Inform the user... we don't know what to do.
-                    JOptionPane.showMessageDialog(PreviewPanel.this,
-                            Localization.lang("Could not print preview") + ".\n" + e.getMessage(),
-                            Localization.lang("Print entry preview"), JOptionPane.ERROR_MESSAGE);
-                    LOGGER.info("Could not print preview", e);
-                }
-            });
-        }
+    public void print() {
```

Comment:
This method does not work on my machine. In the swing version a dialog would appear. Here, the printer icon pops up in the taskbar, but no documents are added.

Reply:
ohh gosh. By default, I print to PDF and thus the PDF print dialog poped up and everything seemed to work just fine. But, of course, in general you want to have a confirmation dialog and a possibility to change the printer. Luckily, javafx provides a default dialog for printing, so this was easy to implement.